INDEXING.md

```diff
@@ -4,48 +4,29 @@
+* An object is part of a batch of objects queued for bulk indexing within Argo (deprecated)
 
-In both cases, Argo uses dor-services to:
+To support indexing, dor-services will:
 * load the object from Fedora.
 * create a Solr document representing the object (using its `to_solr` method).
 * save the document to the Solr instance specified in the configuration, possibly committing immediately.
 
-
 ## Argo::Indexer
 
+This is deprecated for `Dor::IndexingService` in `dor-services`.
+
 ### Reusable methods
 
 The `Argo::Indexer` class provides re-usable methods for indexing:
 * `Argo::Indexer.reindex_pid` takes a PID (druid) and reindexes it as per the above explanation (load from Fedora, create Solr doc, save to Solr).  It also takes an optional logger to use instead of the default (which might be useful for something like logging info about a request triggering the reindex from in the Rails app).  Finally, it gives the option to swallow exceptions.  The default is to let them bubble up, but a caller doing bulk indexing might wish to proceed and index everything it can, with the expectation that the logs will be reviewed later for errors (though note that only `StandardError` errors get trapped, all others always propogate).  The outcome of each indexing attempt is logged, with the PID (druid) and status of the attempt.
 * `Argo::Indexer.generate_index_logger` will create a logger instance that writes to the file specified in `Settings.INDEXER.LOG` with leading string obtained by executing the block provided to `generate_index_logger`.
 
-### Example usage by DorController (which backs the `/dor/reindex` endpoint in Argo)
-
-`DorController.index_logger` uses `Argo::Indexer.generate_index_logger` to generate a logger that leads with the request ID, since those logging events will be triggered from a web request:
-```ruby
-  def index_logger
-    @index_logger ||= Argo::Indexer.generate_index_logger { request.uuid }
-  end
-```
-
-`DorController.reindex` calls `Argo::Indexer.reindex_pid` with the logger instantiated by `index_logger`, and asks Solr to commit the added document (since calling code may expect the update to be available immediately).  Since it uses the default behavior of letting errors bubble up, it specifically deals with `ActiveFedora::ObjectNotFoundError`, rendering a 404 response.
-```ruby
-  def reindex
-    solr_doc = Argo::Indexer.reindex_pid params[:pid], index_logger
-    Dor::SearchService.solr.commit
-    render :plain => "Status:ok<br> Solr Document: #{solr_doc.inspect}"
-  rescue ActiveFedora::ObjectNotFoundError # => e
-    render :status => 404, :plain => 'Object does not exist in Fedora.'
-    return
-  end
-```
-
-
 ## Bulk reindexing
 
+This method is deprecated. See the [`dor_indexing_app`](https://github.com/sul-dlss/dor_indexing_app) Wiki for the new methods. The `Argo::PidGatherer`, however, is not migrated so if you need to gather pids directly from Fedora, refer to that code.
+
```

Comment:
alright, so that's a really big performance improvement, and might totally alleviate the issues with doing a full reindex using sulmq.  i think the last time i used sulmq to do a full reindex was in late 2014, and it took something like 3.5 days to do 400k objects.  i'd be happy to try the new improved sulmq and dor_indexing_app setup instead of the argo reindexing workers if/when the time comes for another full reindex (or a reindex of a significant chunk of the objects in DOR).

and actually, while we'd love to use the `PidGatherer` mechanism, that doesn't provide reliable results, because, for whatever reason, the Fedora resource index (used via `risearch`) doesn't return complete results.  see #380.  as such, we still have to dump a list of pids from fedora's mysql db if we want to index from scratch and we don't have/trust a current solr index.  the script for doing that isn't called out anywhere in the top level documentation, but it lives here, and has decent usage comments at the top of it (lives in the argo codebase git repo, but is run on the DOR machine):
https://github.com/sul-dlss/argo/blob/master/bin/dump_fedora_pids.rb

anyway, discussed for a bit yesterday afternoon w/ @cbeer and @eefahy.  as i understood it, we all agreed that
* we'd like `PidGatherer`/`risearch` to work, but probably won't have the time to actually investigate why the resource index doesn't return complete results.  if we can't get it to work, maybe we should shut it off?
* full index rebuilds (or rebuilds of a significant chunk of the index) tend to be exceptional situations, and the most helpful thing is to have complete documentation of what works and what doesn't, and how things fit together, since it's hard to have reliable readymade solutions for unexpected circumstances.

so, i propose something like the following text:

> This approach is deprecated, but we haven't yet ironed out a process for doing a full reindex from scratch.  One alternate option to this approach would be to queue the IDs of the objects that need to be reindexed using sulmq, for processing by dor_indexing_app.  See the [`dor_indexing_app`](https://github.com/sul-dlss/dor_indexing_app) Wiki for the new methods. The `Argo::PidGatherer`, is not migrated, but we've had issues with our Fedora instance's `risearch` results (see #380).  As such, the following script may be useful in obtaining a full PID list from DOR, in the event that there's no trustworthy Solr index from which to get a list of IDs:
> https://github.com/sul-dlss/argo/blob/master/bin/dump_fedora_pids.rb

@eefahy had suggested improving the `dump_fedora_pids.rb` approach by having it automatically SCP the results to the box where the PIDs will be fed to sulmq.

anyway, that's my $0.02.  no strong objection to merging as-is, if you'd still prefer to do that.